### ISSUE TITLE
-A --allow should take precedence over -D --deny in CLI

### ISSUE THREAD

**Issue by tyranron** [rust-lang/rust-clippy#4778]
I have a weird issue, where I cannot disable a single lint via command line.

Running the following
```bash
cargo +beta clippy --all -- -D clippy::pedantic -D warnings \
	                    -A clippy::must_use_candidate
```
still results in lint being applied:
```
error: this method could have a `#[must_use]` attribute
  --> crates/tracerr/src/trace.rs:50:5
   |
50 |     fn deref(&self) -> &Vec<Frame> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: add the attribute: `#[must_use] fn deref(&self) -> &Vec<Frame>`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#must_use_candidate
```

Changing the order of the CLI arguments doesn't affect anything.

While disabling lint via `!#[allow(clippy::must_use_candidate)]` attribute in crate's root works OK, as expected.

I suggest that things listed in `allow` (no matter where) as a _white list_ should take precedence over `deny` as a _black list_.

```bash
❯ cargo +beta clippy -- --version
clippy 0.0.212 (c8e3cfbd 2019-10-28)
```

**Comment by flip1995** [comment]
The logic of handling command line lint flags is in rustc, not in Clippy:
https://github.com/rust-lang/rust/blob/61a551b4939ec1d5596e585351038b8fbd0124ba/src/librustc/lint/levels.rs#L68-L75

To check if this is a Clippy issue: What is the result of `cargo check -- -D unused -A dead_code`?
